runtime/src/main/java/org/corfudb/protocols/wireprotocol/TokenRequest.java

```diff
@@ -78,6 +81,17 @@ public TokenRequest(Long numTokens, List<UUID> streams) {
         txnResolution = null;
     }
 
+    /**
+     * Constructor for generating an Ordered Globally Unique Id TokenRequest.
+     * {@link TokenRequest::TK_OGUID}
+     *
+     */
+    public TokenRequest(Long numTokens) {
+        this.reqType = TK_OGUID;
+        this.numTokens = numTokens;
+        this.streams = null;
```

Comment:
is it possible to avoid using null-s here?

Reply:
It's required for the serializer.

Reply:
You still can use Optional, let's try to implement that tomorrow

Reply:
Added comment instead as we discussed.